examples/crm/src/sales/SalesEdit.tsx

```diff
@@ -0,0 +1,49 @@
+import {
+    BooleanInput,
+    DeleteWithConfirmButton,
+    Edit,
+    SaveButton,
+    SimpleForm,
+    Toolbar,
+    useGetIdentity,
+    useRecordContext,
+} from 'react-admin';
+import { Sale } from '../types';
+import { SalesForm } from './SalesForm';
+import { useIsAdmin } from './useIsAdmin';
+
+function EditToolbar() {
+    const { identity } = useGetIdentity();
+    const record = useRecordContext<Sale>();
+
+    return (
+        <Toolbar sx={{ display: 'flex', justifyContent: 'space-between' }}>
+            <SaveButton />
+
+            <DeleteWithConfirmButton disabled={record?.id === identity?.id} />
+        </Toolbar>
+    );
+}
+
+export function SalesEdit() {
+    const { identity } = useGetIdentity();
+    const record = useRecordContext<Sale>();
+
+    const isAdmin = useIsAdmin();
+    if (!isAdmin) {
+        return null;
+    }
+
+    return (
+        <Edit>
+            <SimpleForm toolbar={<EditToolbar />}>
+                <SalesForm />
+
+                <BooleanInput
```

Comment:
<!-- Notes are always non-blocking and simply highlight something the reader should take note of.  -->
**note (non-blocking):** I don't think this readOnly props work. When i'm logged as Jane Doe, i can update my status from this page...‏